protocols/protocol/cloudstate/entity.proto

```diff
@@ -27,10 +27,25 @@ import "google/protobuf/descriptor.proto";
 option java_package = "io.cloudstate.protocol";
 option go_package = "cloudstate/protocol";
 
+message Metadata {
+    repeated MetadataEntry entries = 1;
+}
+
+message MetadataEntry {
```

Comment:
Document as to what the behavior/semantics of this is? Are duplicate keys allowed? What happens if they are? :)

Reply:
Behavior/semantics of metadata are undefined at the Cloudstate protocol level, this is for passing transport specific metadata, and so correspondingly the semantics are 100% dependent on whichever transport this metadata has come from or is going to.

Reply:
I've added detailed docs to the spec on this now.